src/index/api.rs

```diff
@@ -26,147 +33,132 @@ pub(crate) struct CrateOwner {
     pub(crate) name: String,
 }
 
-impl RegistryCrateData {
-    pub(crate) fn get_from_network(pkg: &MetadataPackage) -> Result<Self> {
-        let (release_time, yanked, downloads) = get_release_time_yanked_downloads(pkg)?;
-        let owners = get_owners(pkg)?;
+impl Api {
+    pub(super) fn new(api_base: Option<Url>) -> Result<Self> {
+        let headers = vec![
+            (USER_AGENT, HeaderValue::from_static(APP_USER_AGENT)),
+            (ACCEPT, HeaderValue::from_static("application/json")),
+        ]
+        .into_iter()
+        .collect();
+
+        let client = reqwest::blocking::Client::builder()
+            .default_headers(headers)
+            .build()?;
+
+        Ok(Self { api_base, client })
+    }
 
-        Ok(Self {
+    fn api_base(&self) -> Result<Url> {
+        self.api_base
+            .clone()
+            .ok_or_else(|| err_msg("index is missing an api base url"))
+    }
+
+    pub(crate) fn get_crate_data(&self, name: &str, version: &str) -> RegistryCrateData {
+        let (release_time, yanked, downloads) = self
+            .get_release_time_yanked_downloads(name, version)
+            .unwrap_or_else(|err| {
+                info!("Failed to get crate data for {}-{}: {}", name, version, err);
```

Comment:
```suggestion
                warn!("Failed to get crate data for {}-{}: {}", name, version, err);
```

Reply:
I wasn't sure about whether this should be a warning given that not all registries might support it, but just 2 warning logs per build seems fine.